app-modules/student-data-model/tests/Student/ListStudentsTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use AdvisingApp\StudentDataModel\Filament\Resources\StudentResource\Pages\ListStudents;
+use AdvisingApp\StudentDataModel\Models\Student;
+use App\Models\User;
+
+use function Pest\Laravel\actingAs;
+use function Pest\Livewire\livewire;
+
+it('can filter students by first generation', function () {
+    Student::truncate();
+
+    $user = User::factory()->licensed(Student::getLicenseType())->create();
+
+    $user->givePermissionTo('student.view-any');
+    $user->givePermissionTo('student.*.view');
+
+    actingAs($user);
+
+    $students = Student::factory()->count(5)->create();
+
+    livewire(ListStudents::class)
+        ->assertCanSeeTableRecords($students)
+        ->filterTable('firstgen', true)
+        ->assertCanSeeTableRecords($students->where('firstgen', true))
+        ->assertCanNotSeeTableRecords($students->where('firstgen', false))
+        ->filterTable('firstgen', false)
+        ->assertCanSeeTableRecords($students->where('firstgen', false))
+        ->assertCanNotSeeTableRecords($students->where('firstgen', true));
```

Comment:
There is too much randomness in this test. You should create a set group of Students with first_gen and a set group without. Then test the filter using those groups.

You can end up in a scenario here with no users in either scenario so then the test assertion is worthless.

Reply:
Done